src/Storages/StorageFile.cpp

```diff
@@ -581,32 +592,54 @@ void StorageFile::truncate(
 
 void registerStorageFile(StorageFactory & factory)
 {
+    StorageFactory::StorageFeatures storage_features{
+        .supports_settings = true,
+        .source_access_type = AccessType::FILE
+    };
+
     factory.registerStorage(
         "File",
-        [](const StorageFactory::Arguments & args)
+        [](const StorageFactory::Arguments & factory_args)
         {
-            ASTs & engine_args = args.engine_args;
+            StorageFile::CommonArguments storage_args{
+                .table_id = factory_args.table_id,
+                .columns = factory_args.columns,
+                .constraints = factory_args.constraints,
+                .context = factory_args.context
+            };
 
-            if (!(engine_args.size() >= 1 && engine_args.size() <= 3)) // NOLINT
+            ASTs & engine_args_ast = factory_args.engine_args;
+
+            if (!(engine_args_ast.size() >= 1 && engine_args_ast.size() <= 3)) // NOLINT
                 throw Exception(
                     "Storage File requires from 1 to 3 arguments: name of used format, source and compression_method.",
                     ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
 
-            engine_args[0] = evaluateConstantExpressionOrIdentifierAsLiteral(engine_args[0], args.local_context);
-            String format_name = engine_args[0]->as<ASTLiteral &>().value.safeGet<String>();
+            engine_args_ast[0] = evaluateConstantExpressionOrIdentifierAsLiteral(engine_args_ast[0], factory_args.local_context);
+            storage_args.format_name = engine_args_ast[0]->as<ASTLiteral &>().value.safeGet<String>();
 
-            String compression_method;
-            StorageFile::CommonArguments common_args{
-                args.table_id, format_name, compression_method, args.columns, args.constraints, args.context};
+            if (factory_args.storage_def->settings)
+            {
+                Context local_context_copy = factory_args.local_context;
```

Comment:
I'm not sure if we should use `local_context` here, because on server startup it's actually a copy of global context, so storage may behave differently after server restart. If we use `local_context` to take into account profile, user and session settings, then we probably should add all changed settings to CREATE query and write it to metadata file.

Reply:
Maybe we can use global context here? It would mean that we use server-wide defaults + settings from the query itself. This means that the behavior of table changes if you change server-wide settings, but this is also seems to work like this now.

Not sure if using the current user and session settings when creating a table is a good behavior, and also it would require more code -- currently we can't tell which part of `FormatSettings` was changed from default.

Reply:
While I'm at it, I should probably rename `CommonArguments.context` to `global_context` or whatever it is -- having `local_context` and "just some unspecified `context`" is always confusing.

Reply:
I'm ok with global context.

Reply:
I also think that at create time it should pick global context, ignoring local user setting.

Otherwise it can start misbehaving after server restart.